packages/api/src/domain/medical/patient-demographics.ts

```diff
@@ -0,0 +1,491 @@
+      const linkedDobSplit = splitDob(linkDemographics.dob);
+      const overlappingDateParts = linkedDobSplit.filter(dp => patientDobSplit.includes(dp));
+      if (overlappingDateParts.length >= 2) {
+        score += 2;
+        matchedFields.dob = linkDemographics.dob;
+      }
+    }
+  }
+  if (coreDemographics.gender && linkDemographics.gender) {
+    // Gender exact match
+    if (coreDemographics.gender === linkDemographics.gender) {
+      score += 1;
+      matchedFields.gender = linkDemographics.gender;
+    } else {
+      // Gender approximate match
+      // TODO
+    }
+  }
+  // Names exact match
+  const overLapNames = linkDemographics.names.filter(name => coreDemographics.names.includes(name));
+  if (overLapNames.length > 0) {
+    score += 10;
+    matchedFields.names = overLapNames;
+  } else {
+    // Names approximate match
+    // TODO
+  }
+  // Address exact match
+  const overLapAddress = linkDemographics.addresses.filter(a =>
+    coreDemographics.addresses.includes(a)
+  );
+  if (overLapAddress.length > 0) {
+    score += 2;
+    matchedFields.addresses = overLapAddress;
+  } else {
+    // Address approximate match
+    // TODO
+    // Current address zip / city match
+    // TODO Mark the first address as current in Dash
+    if (coreDemographics.addresses.length > 0) {
+      const currentPatientAddress = JSON.parse(coreDemographics.addresses[0]) as LinkGenericAddress;
+      const linkDemograhpicsAddesses = linkDemographics.addresses.map(
+        address => JSON.parse(address) as LinkGenericAddress
+      );
+      if (
+        linkDemograhpicsAddesses.map(address => address.city).includes(currentPatientAddress.city)
+      ) {
+        score += 0.5;
+        matchedFields.addresses = overLapAddress;
+      }
+      if (
+        linkDemograhpicsAddesses.map(address => address.zip).includes(currentPatientAddress.zip)
+      ) {
+        score += 0.5;
+        matchedFields.addresses = overLapAddress;
+      }
+    }
+  }
+  // Telephone exact match
+  const overLapTelephone = linkDemographics.telephoneNumbers.filter(tn =>
+    coreDemographics.telephoneNumbers.includes(tn)
+  );
+  if (overLapTelephone.length > 0) {
+    score += 2;
+    matchedFields.telephoneNumbers = overLapTelephone;
+  } else {
+    // Telephone approximate match
+    // TODO
+  }
+  // Email exact match
+  const overLapEmail = linkDemographics.emails.filter(e => coreDemographics.emails.includes(e));
+  if (overLapEmail.length > 0) {
+    score += 2;
+    matchedFields.emails = overLapEmail;
+  } else {
+    // Email approximate match
+    // TODO
+  }
+  if (linkDemographics.ssns.length > 0) scoreThreshold = 21;
+  const overLapSsn = linkDemographics.ssns.filter(ssn => coreDemographics.ssns.includes(ssn));
+  // SSN exact match
+  if (overLapSsn.length > 0) {
+    score += 5;
+    matchedFields.ssns = overLapSsn;
+  } else {
+    // SSN approximate match
+    // TODO
+  }
+  const isMatched = true || score >= scoreThreshold;
```

Comment:
Intentional that `true ||`?

Reply:
wrong file in the commit -- need to get this thing out before any else goes wrong!

Reply:
Fixed.